src/luxonis_ml/data/dataset.py

```diff
@@ -701,7 +696,13 @@ def _add_execute(self, additions=None, transaction_to_additions=None):
                 if additions is None or transaction_to_additions is None:
                     raise Exception("additions and transaction_to_additions required for adding data")
 
-                addition = additions[transaction_to_additions[transaction['_id']]]
+                if transaction_to_additions.get(str(transaction['_id'])) is None:
+                    # TODO: find a better way to handle those
+                    print(f"Cannot find transaction with id: {str(transaction['_id'])}")
+                    print(f"Seems like a failed transaction?")
+                    continue
```

Comment:
Let's wait to see if this is still an issue after adding some rollback features because I also encountered this when I would get errors in `_add_filter` etc. and there isn't proper error handling of the transactions